src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -51,7 +56,7 @@ public override RazorCSharpDocument WriteDocument(RazorCodeDocument codeDocument
             cSharp,
             _options,
             allOrderedDiagnostics.ToArray(),
-            context.SourceMappings.ToArray(),
+            context.SourceMappings.ToImmutableArray(),
```

Comment:
Tiny nit: Call the `ToImmutable()` instance method on a builder, to avoid an extra null check.